### PR TITLE
Manage Promotions Page

### PR DESCRIPTION
**Pull request by nssafir**
Added a manage promotions page. Displays current promotions and allows admin to add new promotions.

### CODE REVIEW

File: backend/api/models/Promotion.model.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
 const { Schema, model } = require('mongoose')
     
 const promoSchema = Schema({
-    isbn: { type: Number, required: true },
```
Comment (hunterwilkins2) [inline backend/api/models/Promotion.model.js:4]: Can you add a start date? You can just set the default value to Date.now(). Also I think you can remove the title.

File: backend/api/models/Promotion.model.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
     discount: { type: Number, required: true },
 })
-    
+
+/*
 promoSchema.index({
-    isbn: 'text',
     title: 'text',
 })
+*/
```
Comment (hunterwilkins2) [inline backend/api/models/Promotion.model.js]: Probably don't need promoSchema.index

File: backend/api/routes/promotions.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    try {
+        const { date, 
+            title, 
+            discount } = req.body
+
+        const promotion = new Promotion({
+            endDate: date,
+            title: title,
+            discount: discount })
+
```
Comment (hunterwilkins2) [inline backend/api/routes/promotions.js:30]: Update post route to include the optional start date and remove the title

**Review by hunterwilkins2 — CHANGES_REQUESTED** [review]

File: backend/api/models/Promotion.model.js (hunk 1 above)
Comment (nssafir) [reply backend/api/models/Promotion.model.js:4]: @galaxyproduction I can add a startdate -- for the title my thought was customers will actually have to type the promotion in at some point.

Comment (hunterwilkins2) [reply backend/api/models/Promotion.model.js:4]: Ohhh. I thought it was the title of a book. Okay. Nvm on that one

**Review by hunterwilkins2 — APPROVED** [review]
Looks good! I'll merge these changes in

**Pull request merged**